go/vt/sqlparser/parse_test.go

```diff
@@ -4472,50 +5205,73 @@ var incorrectlyParseForNonSelect = []string{
 }
 
 // TestKeywordsCorrectlyParse ensures that certain keywords can be parsed by a series of edit queries.
-func TestKeywordsCorrectlyParse(t *testing.T) {
+func TestKeywordsCorrectlyDoParse(t *testing.T) {
 	aliasTest := "SELECT 1 as %s"
 	iTest := "INSERT INTO t (%s) VALUES (1)"
 	dTest := "DELETE FROM t where %s=1"
 	uTest := "UPDATE t SET %s=1"
 	cTest := "CREATE TABLE t(%s int)"
 
+	// Define expected format for non-reserved keywords
+	aliasTestFmt := "select 1 as %s from dual"
+	iTestFmt := "insert into t(%s) values (1)"
+	dTestFmt := "delete from t where %s = 1"
+	uTestFmt := "update t set %s = 1"
+	cTestFmt := "create table t (\n\t%s int\n)"
+
+	// TODO: most of them aren't back-ticked...but shouldn't they all be?
```

Comment:
This test is probably superfluous, anything should be an acceptable identifier if you backtick quote it. The real test is which identifiers require backtick quoting, and making sure that list matches MySQL

Reply:
This is testing if our parser adds backquotes to the keywords when we don't explicitly add them in the query. Oddly, a little more than half of the words in the list of "non-reserved in MySQL keywords" don't get wrapped in backquotes.

Reply:
We don't actually care about that. Any statement we generate should backtick quote all identifiers. This functionality is only used in tests to verify that we parsed what we think we did.